### PR TITLE
hotfix: updating dependencies for security vulnerabilities #45

### PR DESCRIPTION
**Pull request by Shahar-Y**
Should fix security vulnerabilities 
#45

### CODE REVIEW

File: src/file/file.service.spec.ts [hunk 1]
```diff
@@ -34,11 +34,10 @@ const testUpload = {
 describe('File Logic', () => {
 
   before(async () => {
-
     // Remove files from DB
     const removeCollectionPromises = [];
     for (const i in mongoose.connection.collections) {
-      removeCollectionPromises.push(mongoose.connection.collections[i].remove({}));
+      removeCollectionPromises.push(mongoose.connection.collections[i].deleteOne({}));
```
Comment (omrishtam) [inline src/file/file.service.spec.ts]: should be `deleteMany`

File: src/file/upload.model.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ uploadSchema.post('save', handleE11000);
 uploadSchema.post('update', handleE11000);
 uploadSchema.post('findOneAndUpdate', handleE11000);
 
-uploadSchema.post('save', (error, doc, next) => {
+uploadSchema.post('save', (error: any, doc: any, next: any) => {
```
Comment (omrishtam) [inline src/file/upload.model.ts]: try using types for the parameters.

File: src/file/file.model.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ fileSchema.post('update', handleE11000);
 fileSchema.post('findOneAndUpdate', handleE11000);
 fileSchema.post('insertMany', handleE11000);
 
-fileSchema.post('save', (error, doc, next) => {
+fileSchema.post('save', (error: any, doc: any, next: any) => {
```
Comment (omerzamir) [inline src/file/file.model.ts]: Do not use type any, Check if there are types exist.

File: src/server.ts [hunk 4]
```diff
@@ -56,11 +56,14 @@ export class Server {
...
-    await mongoose.connect(`mongodb://${mongoHost}:${config.db.port}/${config.db.name}`);
+    await mongoose.connect(
+      `mongodb://${mongoHost}:${config.db.port}/${config.db.name}`,
+      { useCreateIndex: true, useNewUrlParser: true, useFindAndModify: false }
+    );
     const db = mongoose.connection;
     db.on('error', console.error.bind(console, 'connection error:'));
     db.once('open', () => {
-      console.log('DB IS CONNECTED!');
+      console.log(`DB IS CONNECTED! to mongodb://${mongoHost}:${config.db.port}/${config.db.name}`);
```
Comment (omerzamir) [inline src/server.ts]: Remove console.log

Comment (omerzamir) [reply src/server.ts]: +1

File: src/utils/spec.helper.ts [hunk 5]
```diff
@@ -43,7 +43,10 @@ before(async () => {
   console.log(process.env.MONGO_HOST);
   const mongoHost = process.env.MONGO_HOST || config.db.host;
   console.log(mongoHost);
-  await mongoose.connect(`mongodb://${mongoHost}:27017/file`);
+  await mongoose.connect(
```
Comment (omrishtam) [inline src/utils/spec.helper.ts]: Delete the `console.log` above.

File: src/server.ts [hunk 6]
```diff
@@ -56,11 +56,13 @@ export class Server {
...
   private async connectDB() {
     const mongoHost = process.env.MONGO_HOST || config.db.host;
-    await mongoose.connect(`mongodb://${mongoHost}:${config.db.port}/${config.db.name}`);
+    await mongoose.connect(
+      `mongodb://${mongoHost}:${config.db.port}/${config.db.name}`,
+      { useCreateIndex: true, useNewUrlParser: true, useFindAndModify: false }
+    );
     const db = mongoose.connection;
     db.on('error', console.error.bind(console, 'connection error:'));
     db.once('open', () => {
```
Comment (omerzamir) [inline src/server.ts]: It does nothing.

**Comment by Shahar-Y** [comment]
Closing for pr to develop

**Pull request closed without merging**